src/tests/mocks/github.rs

```diff
@@ -1,21 +1,83 @@
 use octocrab::Octocrab;
+use std::collections::HashMap;
+use std::time::Duration;
 use wiremock::MockServer;
 
 use crate::create_github_client;
+use crate::github::GithubRepoName;
 use crate::tests::mocks::app::{default_app_id, AppHandler};
-use crate::tests::mocks::repository::RepositoriesHandler;
+use crate::tests::mocks::comment::Comment;
+use crate::tests::mocks::repository::{mock_repo, mock_repo_list};
 use crate::tests::mocks::World;
 
+const DEFAULT_TIMEOUT: Duration = Duration::from_secs(3);
+
+/// Represents the state of a simulated GH repo.
+pub struct GitHubRepoState {
+    // We store comments from all PRs inside a single queue, because
+    // we don't necessarily know beforehand which PRs will receive comments.
+    comments_queue: tokio::sync::mpsc::Receiver<Comment>,
+    // We need a local queue to avoid skipping comments received out of order.
+    pending_comments: Vec<Comment>,
+}
+
+impl GitHubRepoState {
+    /// Wait until a comment from the given pull request was received.
+    /// If a comment from a different PR is received, it is inserted into
+    /// `pending_comments`, to be picked up later.
+    async fn get_comment(&mut self, pr: u64) -> Comment {
+        // First, try to resolve the comment from the pending comment list
+        if let Some(index) = self.pending_comments.iter().position(|c| c.pr == pr) {
+            return self.pending_comments.remove(index);
+        }
+        // If it is not there, wait until some comment is received
+        loop {
+            let comment = self
+                .comments_queue
+                .recv()
```

Comment:
Just to confirm, if the sender panic this loop will be broke out, right?

Reply:
Good question. If all senders (well, there should be only one) panic, they should be unwound, therefore their Drop impls will run, which should drop the sender, and that should result in this `recv()` call returning `None`. If something went bad, the call of this method is wrapped in a `timeout`, so hopefully it shouldn't last more than a few seconds until we saw that in the test that something is wrong.